File: __tests/RendererContextTest.js
module("Renderer context tests");
//
var canvasElement = document.createElement('canvas');
var RendererContext = jscore.RendererContext;
var ctx = new RendererContext({
    canvasElement: canvasElement
});
//
test("Renderer context scale", function () {
    //scale = 1
    ok(ctx.getProperty('NOTE_HEAD_HEIGHT') === 10, 'NOTE_HEAD_HEIGHT (scale = 1)');
    ok(ctx.getProperty('NOTE_HEAD_WIDTH') === 15, 'NOTE_HEAD_WIDTH (scale = 1)');
    ctx.setScale(2);
    ok(ctx.getProperty('NOTE_HEAD_HEIGHT') === 20, 'NOTE_HEAD_HEIGHT (scale = 2)');
    ok(ctx.getProperty('NOTE_HEAD_WIDTH') === 30, 'NOTE_HEAD_WIDTH (scale = 2)');
    ctx.setScale(0.5);
    ok(ctx.getProperty('NOTE_HEAD_HEIGHT') === 5, 'NOTE_HEAD_HEIGHT (scale = 0.5)');
    ok(ctx.getProperty('NOTE_HEAD_WIDTH') === parseInt(15/2), 'NOTE_HEAD_WIDTH (scale = 0.5)');
    ctx.setScale(1);
});

File: __tests/java/RendererContextTest.java
/**
 * *************************************
 */
package com.mzm.score.test;

import com.mzm.score.RendererContext;
import com.mzm.score.bar.note.Beam;
import com.mzm.score.bar.note.Head;
import com.mzm.score.bar.note.Key;
import com.mzm.score.bar.note.KeyName;
import com.mzm.score.bar.note.Note;
import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author maycon.mellos
 */
public class RendererContextTest {

    public static void run() throws Exception {
        RendererContext ctx = RendererContext.getInstance();
        //scale = 1
        Test.test(ctx.getProperty(RendererContext.Property.NOTE_HEAD_HEIGHT) == 10);
        Test.test(ctx.getProperty(RendererContext.Property.NOTE_HEAD_WIDTH) == 15);
        ctx.setScale(2);
        Test.test(ctx.getProperty(RendererContext.Property.NOTE_HEAD_HEIGHT) == 20);
        Test.test(ctx.getProperty(RendererContext.Property.NOTE_HEAD_WIDTH) == 30);
        ctx.setScale(0.5);
        Test.test(ctx.getProperty(RendererContext.Property.NOTE_HEAD_HEIGHT) == 5);
        Test.test(ctx.getProperty(RendererContext.Property.NOTE_HEAD_WIDTH) == 15/2);
        ctx.setScale(1);
    }
}
